src/measure_theory/lp_space.lean

```diff
@@ -1218,4 +1217,432 @@ by { rw eq, exact continuous_pos_part.comp continuous_neg }
+    = (∫⁻ a, filter.at_top.liminf (λ (m : ℕ), (nnnorm (f n a - f m a) : ℝ≥0∞)^p) ∂μ),
+  { rw [snorm', h_no_pow], },
+  refine lintegral_congr_ae (h_lim.mono (λ a ha, _)),
+  rw filter.tendsto.liminf_eq,
+  simp_rw [ennreal.coe_rpow_of_nonneg _ (le_trans zero_le_one hp1), ennreal.tendsto_coe],
+  refine ((nnreal.continuous_rpow_const (le_trans zero_le_one hp1)).tendsto
+      (nnnorm (f n a - f_lim a))).comp _,
+  exact (continuous_nnnorm.tendsto (f n a - f_lim a)).comp (tendsto_const_nhds.sub ha),
+end
+
+/-! ### `Lp` is complete iff Cauchy sequences of `ℒp` have limits in `ℒp` -/
+
+lemma tendsto_Lp_of_tendsto_ℒp [hp : fact (1 ≤ p)] {f : ℕ → Lp E p μ}
+  (h_tendsto : ∃ (f_lim : α → E) (hf_lim_meas : mem_ℒp f_lim p μ),
+    ∀ ε, 0 < ε → ε < ∞ → (∃ N, ∀ n, N ≤ n → snorm (f n - f_lim) p μ < ε)) :
+  ∃ (g : Lp E p μ), filter.at_top.tendsto f (𝓝 g) :=
+begin
+  simp_rw metric.tendsto_at_top,
+  rcases h_tendsto with ⟨f_lim, f_lim_ℒp, h_tendsto⟩,
+  refine ⟨mem_ℒp.to_Lp f_lim f_lim_ℒp, λ ε hε, _⟩,
+  have hε_pos : 0 < ennreal.of_real ε, from ennreal.of_real_pos.mpr hε,
+  cases (h_tendsto (ennreal.of_real ε) hε_pos ennreal.of_real_lt_top) with N h_tendsto,
+  refine ⟨N, λ n hn, _⟩,
+  specialize h_tendsto n hn,
+  have hp_pos : 0 < p, from ennreal.zero_lt_one.trans_le hp.elim,
+  rw [dist_def, ←@ennreal.to_real_of_real ε (le_of_lt hε),
+    ennreal.to_real_lt_to_real _ ennreal.of_real_ne_top],
+  { suffices h_coe : ⇑(f n) - f_lim =ᵐ[μ] ⇑(f n) - ⇑(mem_ℒp.to_Lp f_lim f_lim_ℒp).val,
+      by rwa snorm_congr_ae h_coe at h_tendsto,
+    have h_coe' : f_lim =ᵐ[μ] ⇑(mem_ℒp.to_Lp f_lim f_lim_ℒp),
+      from (mem_ℒp.coe_fn_to_Lp _).symm,
+    refine filter.eventually.mp h_coe' (filter.eventually_of_forall (λ x hx, _)),
+    rw [pi.sub_apply, hx],
+    refl, },
+  { rw snorm_congr_ae (Lp.coe_fn_sub _ _).symm,
+    exact snorm_ne_top _, },
+end
+
+lemma tendsto_ℒp_of_tendsto_Lp [hp : fact (1 ≤ p)] {f : ℕ → Lp E p μ}
+  (h_tendsto : ∃ (g : Lp E p μ), filter.at_top.tendsto f (𝓝 g)) :
+  ∃ (f_lim : α → E) (hf_lim_meas : mem_ℒp f_lim p μ),
+    ∀ ε, 0 < ε → ε < ∞ → (∃ N, ∀ n, N ≤ n → snorm (f n - f_lim) p μ < ε) :=
+begin
+  simp_rw metric.tendsto_at_top at h_tendsto,
+  cases h_tendsto with g h_tendsto,
+  refine ⟨g, Lp.mem_ℒp g, λ ε hε hε_top, _⟩,
+  have hε_pos : 0 < ε.to_real, from ennreal.to_real_pos_iff.mpr ⟨hε, lt_top_iff_ne_top.mp hε_top⟩,
+  cases (h_tendsto ε.to_real hε_pos) with N h_tendsto,
+  refine ⟨N, λ n hn, _⟩,
+  specialize h_tendsto n hn,
+  rwa [dist_def, ennreal.to_real_lt_to_real _ (lt_top_iff_ne_top.mp hε_top)] at h_tendsto,
+  rw [←lt_top_iff_ne_top, snorm_congr_ae (Lp.coe_fn_sub _ _).symm],
+  exact Lp.snorm_lt_top _,
+end
+
+lemma tendsto_Lp_iff_tendsto_ℒp [hp : fact (1 ≤ p)] {f : ℕ → Lp E p μ} :
+ (∃ (g : Lp E p μ), filter.at_top.tendsto f (𝓝 g))
+  ↔ ∃ (f_lim : α → E) (hf_lim_meas : mem_ℒp f_lim p μ),
+    ∀ ε, 0 < ε → ε < ∞ → (∃ N, ∀ n, N ≤ n → snorm (f n - f_lim) p μ < ε) :=
+⟨tendsto_ℒp_of_tendsto_Lp, tendsto_Lp_of_tendsto_ℒp⟩
```

Comment:
Absolutely not. Writing an equivalence felt nice, but I agree that they are not useful.

Reply:
Then I'd remove them, since the argument is already a little bit involved, so there's no need to add something in the middle which is not directly needed for the proof.